### PR TITLE
[WIP] Add detailed git workflow

### LINKED ISSUE #602: Detailed Git workflow
#### Nature of issue?

- [ ] Found a bug
- [x] Existing feature enhancement
- [ ] New feature request

#### New feature details:

Should we add a more detailed git workflow in contributing.md like https://github.com/processing/p5.js/wiki/Development do? New contributors are always confused about the git :)

### PR DESCRIPTION
**Pull request by shinytang6**
Before your pull request is reviewed and merged, make sure you

* [x] there are no linting errors -- `npm run lint`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master. If you're asked to make more changes make sure you rebase onto master then too!
* [x] pull request is descriptively named and links to an issue number, i.e. `Fixes #123`

Thank you! Fixes #602

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -49,27 +49,80 @@ Design proposed and theme changes are present at: [Zeplin](https://scene.zeplin.
...
 # How Can I Contribute?
 
-### First Timers
+## Contributing Code
+
+### Step 1: Set up a working copy on your computer
+
+Firstly you need to set up a local fork of our project, so press the "fork" button in Github. This will create a copy of our project in your Github account.
+
+Then you can clone the project to your local machine using this command：
```
Comment (catarak) [inline contributing.md]: "Then you can clone the project to your local machine by entering the following command into a terminal window:"

File: contributing.md [hunk 2]
```diff
@@ -49,27 +49,80 @@ Design proposed and theme changes are present at: [Zeplin](https://scene.zeplin.
...
+
+### Step 4: Creating a pull request
 
 When you create a [pull request](https://help.github.com/articles/creating-a-pull-request/) for a new fix or feature, be sure to mention the issue number for what you're working on. The best way to do it is to mention the issue like this at the top of your description:
 
     Fixes #333
 
 The issue number in this case is "333." The word *Fixes* is magical; GitHub will automatically close the issue when your pull request is merged.
 
+To create a PR you need to push your branch to the origin remote and then press the buttons on GitHub.
```
[the remark this answers is not in the document]
Comment (catarak) [reply contributing.md]: also if we keep this change the language to 
"To create a pull request (PR), you first need to push your branch to the origin remote, and then press the button on GitHub in your forked repository."

**Comment by catarak** [comment]
@lmccart / @shiffman thoughts on how much git workflow to add? what's the threshold for overkill?

File: contributing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (shinytang6) [reply contributing.md]: I just think that doing so can provide a complete process for the first time contributors :thinking:

**Comment by catarak** [comment]
also now that the Processing fellowships have been announced... also @vijithassar, thoughts?

**Comment by lmccart** [comment]
this workflow looks great, but I think any parts that are just describing git commands could be skipped and replaced with a link to a tutorial. this one seems to cover most of it: https://guides.github.com/activities/forking/

**Comment by lmccart** [comment]
and this is a nice intro github tutorial https://guides.github.com/activities/hello-world/

**Comment by catarak** [comment]
cool, that makes sense to me. @shinytang6 can you remove the git specific parts from this and link to the git documentation from on github?

**Comment by vijithassar** [comment]
[this tutorial](http://oss-watch.ac.uk/resources/pullrequest) strikes me as clear and appropriate for beginners, but the source isn't ideal... really wish GitHub provided better official introductory guides

**Comment by shinytang6** [comment]
Sure！Will do that tomorrow :)

**Comment by shinytang6** [comment]
Hey @lmccart @vijithassar ! For some reasons I don't think these tutorials are friendly enough for newcomers.. These tutorials lack some effective contribution processes, think so?

**Comment by catarak** [comment]
i think the two guides that @lmccart are close to what we want. i think that it would make sense to link to them and fill in the gaps (i.e. talk about having `upstream` and `origin`).

**Comment by catarak** [comment]
thank you, looks great!

**Pull request merged**